### PR TITLE
Initial Commit of Chipping Code

### PR DESCRIPTION
**Pull request by ymt123**
Initial Commit of code that reads a video file and extracts chips in two different ways

### CODE REVIEW

**Comment by ymt123** [comment]
I'm not sure how to test this without a test video (which we don't really have). If someone has a clever idea I'd be happy (see: I'd like to) add some tests.

**Comment by agude** [comment]
I'll take a look at the code. For testing, I think Caltrans cameras should be useable: 

> In general, information presented on this web site, unless otherwise indicated, is considered in the public domain. It may be distributed or copied as permitted by law. However, the State does make use of copyrighted data (e.g., photographs) which may require additional permissions prior to your use. In order to use any information on this web site not owned or created by the State, you must seek permission directly from the owning (or holding) sources. The State shall have the unlimited right to use for any purpose, free of any charge, all information submitted via this site except those submissions made under separate legal contract. The State shall be free to use, for any purpose, any ideas, concepts, or techniques contained in information provided through this site. 
>
> http://www.dot.ca.gov/use.html#ownership

Otherwise, I'll review the code now.

**Comment by ymt123** [comment]
Updates completed

File: pelops/datasets/chipper.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+                 desired_framerate=2):
+        self.current_file = 0
+        self.file_list = file_list
+        self.open_func = open_func
+        self.decoder = decoder
+        self.desired_framerate = desired_framerate
+
+    def __iter__(self):
+        for filename in self.file_list:
+            logger.info('Staring file: %s'%filename)
```
Comment (agude) [inline pelops/datasets/chipper.py]: One last `.format()`

**Pull request merged**